config/initializers/custom_exception_mappings.rb

```diff
@@ -0,0 +1,6 @@
+ActionDispatch::ExceptionWrapper.rescue_responses.merge!(
+  "Pundit::NotAuthorizedError"              => :forbidden,
```

Comment:
Does it return `Insights::API::Common::ErrorDocument` like RbacError returned?

Reply:
Yep, that is handled here: https://github.com/RedHatInsights/insights-api-common-rails/blob/master/lib/insights/api/common/application_controller_mixins/exception_handling.rb

which is mixed in the base ApplicationController.